code/ai/aicode.cpp

```diff
@@ -14682,12 +14705,13 @@ int aas_1(object *objp, ai_info *aip, vec3d *safe_pos)
 		Assert(aip->shockwave_object > -1);
 		object	*ship_objp = &Objects[aip->shockwave_object];
 		if (ship_objp == objp) {
-			aip->shockwave_object = -1;
+			aip->shockwave_object = -1;	// this one was already present in retail
```

Comment:
There are some additional weapon cases which also return 0, but don't seem to reset `shockwave_object`, should those be updated as well?

Reply:
The `shockwave_object` field should be reset when, and only when, an `Avoid_shockwave_` flag is removed, not necessarily when the function returns 0.

There is of course the interesting `//	See if too far away to care about shockwave.` case which returns 0 but does not remove the flag, as the flag is commented out -- even though the equivalent distance case for ship shockwaves removes the flag.  The fact that the line is commented out makes me think Volition had a good reason for not clearing the flag.  I'll give that another test.  There's always the possibility of another separate AI profiles flag for that specific line.

Reply:
It looks like the best course of action is indeed to remove the flag (and reset `shockwave_object`) in that case.